flang/lib/Lower/ConvertVariable.cpp

```diff
@@ -1292,11 +1292,10 @@ void Fortran::lower::mapSymbolAttributes(
           return;
         }
         // local CHARACTER variable
-        mlir::Value len;
-        if (charLen)
-          len = genValue(*charLen);
-        else
-          len = builder.createIntegerConstant(loc, idxTy, sym.size());
+        if (!charLen)
+          fir::emitFatalError(loc,
+                              "local characters must have explicit length");
+        mlir::Value len = genValue(*charLen);
```

Comment:
This same four lines of code appears three times in this file.  Should we have a function for this check?

Reply:
Indeed, shared in a small `genExplicitCharLen` helper.
I also slightly updated the `charLenVariable` (in BoxAnalyzer.h) update to follow ultimate symbols, otherwise this change was problematic with modules/host procedures.